docs(release): Update the release doc for the new process (#1185)
adevinta/spark-android: RELEASING.md

# Releasing

## Pre-release

Before each release we will publish a Snapshot version of it and use it to create a pre release on [Github Release page](https://github.com/adevinta/spark-android/releases)

> During a period of ~2 weeks consumers will be able to test it for bugs that can be fixed before it’s considered stable.

Reviewer: ```suggestion
During a period of ~2 weeks, consumers will be able to test it and report bugs that can be fixed before it’s considered stable.
```

Author: Done in 6730295

> Using Github pre-release feature will allow Spark users to create a hook to notice that a new release has been made and test against it.

Reviewer: ```suggestion
Using Github pre-release feature will allow Spark users to be notified that a new release is being prepared and test against it.
```

Author: Done in 6730295

> For example they could hook to it to make a bot post this changelog in their monitoring slack channel or trigger a CI build to certify that this new release doesn’t break their build.

Reviewer: nit: wording

```suggestion
For example they could setup hooks to post the changelog in their monitoring slack channel or trigger a CI build to validates that this new release doesn’t break their build.
```

Author: Done in https://github.com/adevinta/spark-android/commit/6730295a22b486d089a1d554ba9e1e2add81719b

> 1. First create a draft release that by creating a `*.*.*-SNAPSHOT` version tag.

Reviewer: 1st is always first 🙃 

nit: Could we add a link to this?

```suggestion
1. [Create a draft release](https://github.com/adevinta/spark-android/releases/new?tag=X.Y.Z-SNAPSHOT&title=X.Y.Z-SNAPSHOT&prerelease=1) with a `*.*.*-SNAPSHOT` version tag.
```

Author: Done in https://github.com/adevinta/spark-android/commit/6730295a22b486d089a1d554ba9e1e2add81719b

> 2. Use this new tag to use the change log generation tool from GitHub.

Reviewer: What does this mean?!

```suggestion
2. Click on `Generate release notes` to automatically add all the merged pull requests from this diff and contributors of this release.
```

> 3. Remove logs from Dependabot except if they mention big version upgrade for libraries used by our consumers (like Compose or Kotlin version upgrade).

Reviewer: ```suggestion
@dependabot
3. Remove logs from @dependabot except if they mention big version upgrades for libraries used by our consumers (like Compose or Kotlin).
```

Author: Done in https://github.com/adevinta/spark-android/commit/6730295a22b486d089a1d554ba9e1e2add81719b

> 5. 🪄 Format the logs to be close to the format we use for our [CHANGELOG%20STYLE%20GUIDE](./docs/CHANGELOG%20STYLE%20GUIDE.md).

Reviewer: ```suggestion
5. 🪄 Format the logs to be close to the format we use for our [CHANGELOG STYLE GUIDE](./docs/CHANGELOG STYLE GUIDE.md).
```

not sure if the spaces are automagically replaced

Reviewer: ```suggestion
5. Reformat the changelog to be as close as possible to the format we describe in the [CHANGELOG STYLE GUIDE](./docs/CHANGELOG STYLE GUIDE.md).
```

Author: Done in f54b639

> 5. 🪄 Format the logs to be close to the format we use for our [CHANGELOG%20STYLE%20GUIDE](./docs/CHANGELOG%20STYLE%20GUIDE.md).

Reviewer: I like emojis, but I feel like having only a single one for this entry makes it weired.

Author: I agree, I forgot to remove it as I moved the formatting/styling to another file 🙇

> 6. If we’re satisfied with the draft now release it but as a ⚠️ **pre-release**.

Reviewer: ```suggestion
6. If we’re satisfied with the draft, release it but make sure **`⚠️ Set as a pre-release`** is checked.
```

Author: Done in 6730295

> 7. Wait at ==least 2 weeks== for feedback from consumers on the stability of he release.

Reviewer: Use proper markdown formatting for emphase

```suggestion
7. Wait **at least 2 weeks** for feedback from consumers on the stability of this release.
```

Author: Done in 6730295

8. ???

> 9. Profit! 💹

Reviewer 2: Is this added by mistake?

## Release

> 1. Update the `version` in [gradle.properties](https://github.com/adevinta/spark-android/blob/main/gradle.properties) to a non-`SNAPSHOT`.

Reviewer: Please restore the initial link, the "broken" link you are seeing is only because you are inside a PR

Author: Aaah that's why, I'm reverting them!

Author: Done in https://github.com/adevinta/spark-android/commit/6730295a22b486d089a1d554ba9e1e2add81719b

> 2. Update [CHANGELOG.md](https://github.com/adevinta/spark-android/blob/main/CHANGELOG.md)

Reviewer: same here

Author: Done in https://github.com/adevinta/spark-android/commit/6730295a22b486d089a1d554ba9e1e2add81719b

> - Add the new version section and move the *unreleased changes* into it.
> - Update the links at the end of the page.

Reviewer: Don't trust this 🐰 indentation is required (and you need 3 spaces here)
```suggestion
   - Add the new version section and move the *unreleased changes* into it.
   - Update the links at the end of the page.
```

Author: Fixed in [78a11a2](https://github.com/adevinta/spark-android/pull/1185/commits/78a11a24231d904d83dd69ae74eee3d6d1d67ae0)

   git push origin X.Y.Z
   ```
5. Wait for the [publishing workflow](https://github.com/adevinta/spark-android/actions/workflows/publish.yml) to build and publish the release.

> 6. Update the `version` in [gradle.properties](https://github.com/adevinta/spark-android/blob/main/gradle.properties) to the next `SNAPSHOT` version.

Reviewer: same

Author: Done in https://github.com/adevinta/spark-android/commit/6730295a22b486d089a1d554ba9e1e2add81719b

7. Commit and push the changes to a new PR
  ```bash
  git commit -am "chore: prepare next development version"
  ```
8. Go to [Sonatype Nexus](https://s01.oss.sonatype.org/) to promote ([docs & detailed steps](https://central.sonatype.org/publish/release/))
   ##### If there is a problem! and you wanna dismiss it just hit `Drop`
   #### If all looks good:
    1. `close` then wait till it completes
    2. `release` the artifact
9. Trigger the manual workflow [![📋 Publish Dokka to GitHub Pages](https://github.com/adevinta/spark-android/actions/workflows/dokka.yml/badge.svg)](https://github.com/adevinta/spark-android/actions/workflows/dokka.yml) with the version tag.

> 10. Draft a [new release](https://github.com/adevinta/spark-android/releases/new) with the version tag, add the corresponding [CHANGELOG.md](https://github.com/adevinta/spark-android/blob/main/CHANGELOG.md) entries, and publish it when ready.

Reviewer: same

Author: Done in https://github.com/adevinta/spark-android/commit/6730295a22b486d089a1d554ba9e1e2add81719b
